src/App.jsx

```diff
@@ -15,41 +11,56 @@ class App extends React.Component {
       actor: '',
       permission: '',
       session: '',
-      accountData: {}
+      accountData: {},
+      error: '',
     };
   }
 
-  componentDidMount = async ()  => { 
+  componentDidMount = async () => {
     this.checkIfLoggedIn();
-  }
+  };
 
   checkIfLoggedIn = async () => {
-    const { auth, accountData } = await ProtonSDK.restoreSession();
-    if (auth && auth.actor && auth.permission) {
+    const { auth, accountData, error } = await ProtonSDK.restoreSession();
+    if (error) {
+      this.setErrorState(error);
```

Comment:
Let's use the early return pattern to unnest some of our code. We can set the error and return out of the function if there is an `error` or `auth && auth.actor && auth.permission` is falsey.

```js
if (...) {
  this.setErrorState(error);
  return;
}

this.setLoggedInState(auth.actor, auth.permission, accountData);
```